Modix.Services/Quote/MessageLinkBehavior.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Modix.Services.Quote
+{
+    public class MessageLinkBehavior : BehaviorBase
+    {
+        private static readonly Regex Pattern = new Regex(
+            @"https://(canary\.)?discordapp\.com/channels/(?<GuildId>\d+)/(?<ChannelId>\d+)/(?<MessageId>\d+)",
+            RegexOptions.Compiled | RegexOptions.IgnoreCase | RegexOptions.CultureInvariant);
+
+        public MessageLinkBehavior(DiscordSocketClient discordClient, IServiceProvider serviceProvider)
+            : base(serviceProvider)
+        {
+            DiscordClient = discordClient;
+            Log = serviceProvider.GetRequiredService<ILogger<MessageLinkBehavior>>();
+        }
+
+        private DiscordSocketClient DiscordClient { get; }
+
+        private ILogger<MessageLinkBehavior> Log { get; }
+
+        protected internal override Task OnStartingAsync()
+        {
+            DiscordClient.MessageReceived += OnMessageReceived;
+
+            return Task.CompletedTask;
+        }
+
+        protected internal override Task OnStoppedAsync()
+        {
+            DiscordClient.MessageReceived -= OnMessageReceived;
+
+            return Task.CompletedTask;
+        }
+
+        private async Task OnMessageReceived(SocketMessage message)
+        {
+            if (!(message is SocketUserMessage userMessage) ||
+                !(userMessage.Author is SocketGuildUser || userMessage.Author.IsBot) ||
```

Comment:
Should we exclude bots? Just thinking about a use case where someone might put a message link in a tag or something and expect a quoted reply.

If we do decide to exclude bots, it probably makes sense to exclude webhooks as well.

Reply:
We should probably exclude bots as some combination of commands may result in an infinite loop. Webhooks too, unless needed.

Reply:
Good point, didn't think about that

Reply:
Good catch - updated.